docs/adr/0001-add-generic-fetcher.md

```diff
@@ -0,0 +1,109 @@
+# Add generic fetcher
+
+- Status: proposed
+- Date: 2024-10-30
+
+## Context
+
+The main motivation for this change is to cover use cases of users that need to download arbitrary files that don't fit
+within an established package ecosystem cachi2 could potentially otherwise support. The target audience is users that
+want to use cachi2 to achieve hermetic builds and want an easy way to also include these arbitrary files, that cachi2
+will account for in the SBOM it produces.
+
+## Decision
+
+This change introduces a generic fetcher, an additional cachi2 package manager. This package manager utilizes a custom
```

Comment:
```diff
- This change introduces a generic fetcher, an additional cachi2 package manager.
+ A new package manager for generic artifacts must be introduced.
```

(since this is a design doc, not a change that introduces anything).

Reply:
My thought was that we would discuss here what needs to be changed, and those changes would be merged together with the ADR

Reply:
Changed.